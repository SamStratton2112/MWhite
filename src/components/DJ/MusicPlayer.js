import { Card } from "react-bootstrap";


const MusicPlayer = ({album}) => {
    console.log(album.src)
    return (
        <Card className="center" style={{backgroundColor:' rgb(66, 65, 65,.5)', color:'white', margin: '1em' }}>
            <Card.Img variant="top" src={album.image} alt={album.name} style={{ width: '15em', height: '16em', paddingTop: '1em'}}/>
            <Card.Body>
                <Card.Title style={{fontSize:'1.7em'}} >{album.name}</Card.Title>
                <a href={album.src}><button className="bttn">Listen on Soundcloud</button></a>
            </Card.Body>
        </Card>
    )
};

export default MusicPlayer;