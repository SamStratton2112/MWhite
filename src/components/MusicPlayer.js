import { Card, Button } from "react-bootstrap";


const MusicPlayer = (song) => {
    return (
        <Card style={{ width: '18rem' }}>
            <Card.Img variant="top" src={song.src} alt={song.name} />
            <Card.Body>
                <Card.Title>{song.name}</Card.Title>
                <Card.Text>
                    {song.name}
                </Card.Text>
                <Button variant="primary">Listen on Soundcloud</Button>
            </Card.Body>
        </Card>
    )
};

export default MusicPlayer;