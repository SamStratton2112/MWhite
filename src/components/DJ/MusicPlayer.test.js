import MusicPlayer from './MusicPlayer';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

const testAlbum = {
    image: 'testImg',
    name: 'testName',
    src: 'testSrc'
}

describe(MusicPlayer, () => {
    it("renders without crashing", () => {
        render(<MemoryRouter>
            <MusicPlayer album={testAlbum} />
        </MemoryRouter>);
    });
    it("matches snapshot", () => {
        const { asFragment } = render(<MemoryRouter>
            <MusicPlayer album={testAlbum} />
        </MemoryRouter>);
        expect(asFragment()).toMatchSnapshot();
    });
});