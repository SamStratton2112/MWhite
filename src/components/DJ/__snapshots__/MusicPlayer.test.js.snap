// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MusicPlayer matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="center card"
    style="color: white; margin: 1em; border: solid; border-color: rgb(184, 17, 17);"
  >
    <img
      alt="testName"
      class="card-img-top"
      src="testImg"
      style="width: 15em; height: 16em; padding-top: 1em;"
    />
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
        style="font-size: 1.7em;"
      >
        testName
      </div>
      <a
        href="testSrc"
      >
        <button
          class="bttn"
        >
          Listen on Soundcloud
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
